crates/gen/src/tables/attribute.rs

```diff
@@ -33,42 +36,61 @@ impl Attribute {
             AttributeType::MemberRef(method) => (reader.blob(method.0, 2), reader.blob(self.0, 2)),
         };
 
-        values.read_u16();
-        sig.read_unsigned();
-        let count = sig.read_unsigned();
-        sig.read_unsigned();
+        let prolog = values.read_u16();
+        assert!(prolog == 0x0001, "CustomAttribute Prolog must be 0x0001"); // Required by spec.
 
-        let mut args: Vec<(String, AttributeArg)> = Vec::with_capacity(count as usize);
+        let _this_and_gen_param_count = sig.read_unsigned();
```

Comment:
I gave a lot of stuff in here names as I was attempting to understand what values were being read while going through the ECMA spec, and decided there was no reason to take them back out since it's more descriptive this way, but obviously let me know if you disagree.